### DISCUSSION TITLE
CLIFp and Flashpoint 12

### DISCUSSION THREAD

**Discussion by TheDoomCider** [oblivioncth/CLIFp#64] [Q&A] [answered]
Hi,
I'm having trouble getting CLIFp to work with Flashpoint 12. I keep getting an error about it not being able to find a valid flashpoint installation. I've tried the windows installer, the zipped download and the Linux version of Flashpoint 12 on my Steam Deck. I'm putting CLIFp in the main flashpoint folder like I did with previous versions of flashpoint and I am using the latest version of CLIFp. Is this a bug or am I missing a step here?

**Comment by oblivioncth** [comment]
If you look in your CLIFp.log, do you see something like
> JSON Error: Could not retrieve key 'onDemandImagesCompressed'. The key does not exist.

**Comment by TheDoomCider** [comment]
Yes, actually, I do.

**Reply by oblivioncth** [reply, marked as answer]
Lol, every time some minor issue like this creeps up I get paranoid that Colin from the FP team made some crazy breaking change I'll have to deal with because of the newer live updates, but alas that still has yet to happen.

What's going on is that when FP12 was first released the compressed images feature wasn't completed yet so the initial "12.0" version didn't have it. The underlying library that CLIFp uses tries to read the value for that preference (onDemandImagesCompressed) so when you try to use CLIFp or FIL with the original FP12 it errors out as that value doesn't exist.

Most people update Infinity right off the bat after downloading it so I didn't think of this potential caveat. I just pushed an update to CLIFp that allows it to work with non-updated copies by treating the presence of that setting as optional, if for some reason you do not wish to update; however, simply updating your current copy of FP12 should get you up and running.

**Reply by TheDoomCider** [reply]
No worries! I get it. Thanks for getting back to me so quickly on this. I will give it a try!

**Reply by TheDoomCider** [reply]
Perfect, that fixed it! Thanks again!

**Reply by oblivioncth** [reply]
Software is supposed to work XD, so please don't hesitate to let me know if you have other issues in the future.
